HLTrigger/HLTfilters/test/triggerResultsFilter_by_PathStatus.py

```diff
@@ -0,0 +1,212 @@
+from HLTrigger.HLTfilters.triggerResultsFilter_cfi import triggerResultsFilter as _triggerResultsFilter
+triggerResultsFilter = _triggerResultsFilter.clone( usePathStatus = True )
+
+# accept if 'Path_1' succeeds
+process.filter_1 = triggerResultsFilter.clone(
+    triggerConditions =  ( 'Path_1', ),
+    l1tResults = '',
+    throw = False
+)
+
+# accept if 'Path_2' succeeds
+process.filter_2 = triggerResultsFilter.clone(
+    triggerConditions = ( 'Path_2', ),
+    l1tResults = '',
+    throw = False
+)
+
+# accept if 'Path_3' succeeds
+process.filter_3 = triggerResultsFilter.clone(
+    triggerConditions = ( 'Path_3', ),
+    l1tResults = '',
+    throw = False
+)
+
+# accept if any path succeeds (explicit OR)
+process.filter_any_or = triggerResultsFilter.clone(
+    triggerConditions = ( 'Path_1', 'Path_2', 'Path_3' ),
+    l1tResults = '',
+    throw = False
+)
+
+# accept if 'Path_1' succeeds, prescaled by 2
+process.filter_1_pre = triggerResultsFilter.clone(
+    triggerConditions =  ( '(Path_1) / 15', ),
+    l1tResults = '',
+    throw = False
+)
+
+# accept if 'Path_2' succeeds, prescaled by 10
+process.filter_2_pre = triggerResultsFilter.clone(
+    triggerConditions = ( '(Path_2 / 10)', ),
+    l1tResults = '',
+    throw = False
+)
+
+# accept if any path succeeds, with different prescales (explicit OR, prescaled)
+process.filter_any_pre = triggerResultsFilter.clone(
+    triggerConditions = ( 'Path_1 / 15', 'Path_2 / 10', 'Path_3 / 6', ),
+    l1tResults = '',
+    throw = False
+)
+
+# accept if any path succeeds (wildcard, '*')
+process.filter_any_star = triggerResultsFilter.clone(
+    triggerConditions = ( 'Path_*', ),
+    l1tResults = '',
+    throw = False
+)
+
+# accept if any path succeeds (wildcard, '?')
+process.filter_any_question = triggerResultsFilter.clone(
+    triggerConditions = ( 'Path_?', ),
+    l1tResults = '',
+    throw = False
+)
+
+# accept if any path succeeds (double wildcard, '*_?')
+process.filter_any_starquestion = triggerResultsFilter.clone(
+    triggerConditions = ( '*_?', ),
+    l1tResults = '',
+    throw = False
+)
+
+# accept if all path succeed (explicit AND)
+process.filter_all_explicit = triggerResultsFilter.clone(
+    triggerConditions = ( 'Path_1 AND Path_2 AND Path_3', ),
+    l1tResults = '',
+    throw = False
+)
+
+# wrong path name (explicit)
+process.filter_wrong_name = triggerResultsFilter.clone(
+    triggerConditions = ( 'Wrong', ),
+    l1tResults = '',
+    throw = False
+)
+
+# wrong path name (wildcard)
+process.filter_wrong_pattern = triggerResultsFilter.clone(
+    triggerConditions = ( '*_Wrong', ),
+    l1tResults = '',
+    throw = False
+)
+
+# empty path list
+process.filter_empty_pattern = triggerResultsFilter.clone(
+    triggerConditions = ( ),
+    l1tResults = '',
+    throw = False
+)
+
+# L1-like path name
+process.filter_l1path_pattern = triggerResultsFilter.clone(
+    triggerConditions = ( 'L1_Path', ),
```

Comment:
Hi @fwyzard, thanks for the PR. I just wanted to ask one clarification before signing it.

In the unit test, I see that the path `L1_Path` always returns `true`, but the corresponding `TriggerResultFilter` (i.e. `filter_l1path_pattern`) always returns `false` (for `filter_l1path_pattern`, I see that `m_expression->patterns()` is empty). Is this related to how strings starting with `L1_` are handled in [`TriggerExpressionParser.h`](https://github.com/cms-sw/cmssw/blob/3707fa96313fde4b4ca7b269df402eadff1f53dd/HLTrigger/HLTcore/interface/TriggerExpressionParser.h#L26)? Could you please elaborate a bit? (I admit I don't follow the details of this)